rust_src/src/buffers.rs

```diff
@@ -168,6 +174,49 @@ impl LispBufferRef {
         unsafe { (*self.text).z }
     }
 
+    // Same as the BUF_BYTE_ADDRESS c macro
+    /// Return the address of character at byte position BYTE_POS in buffer BUF.
+    #[inline]
+    pub fn buf_byte_address(self, byte_pos: isize) -> c_uchar {
```

Comment:
Hmmm, it looks like there is a repeated bit of code here:

```
fn pos_with_range(self, pos) -> isize {
    if pos >= self.gpt_byte() {
        self.gap_size()
    } else {
        0
    }
}
```

This can be used here and in the `dec_pos` method below. Likely in other spots too.